File: ruby/easy/remove_duplicates_from_sorted_list.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val)
#         @val = val
#         @next = nil
#     end
# end

# @param {ListNode} head
# @return {ListNode}
def delete_duplicates(head)
    return nil unless head
    current, runner = head, head.next
    while(runner)
        if current.val != runner.val
            current.next = runner
            current = current.next
        end
        runner = runner.next
    end
    current.next = nil if current
    return head
end

File: javascript/easy/removeDuplicatesFromSortedList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    if(head == null) return null;
    let current = head, runner = head.next;
    while(runner !== null) {
        if(current.val != runner.val) {
            current.next = runner;
            current = runner;
        }
        runner = runner.next;
    }
    current.next = null;
    return head;
};
